assets/scss/3-components/4-modal.scss

```diff
@@ -0,0 +1,73 @@
+#modal {
+  position: fixed;
+  width: 100%;
+  height: 100%;
+  z-index: 11;
+  top: 0;
+  left: 0;
+  @include flex(row, center, center, wrap);
+  opacity: 0;
+  visibility: hidden;
+  transition: all 0.1s ease 500ms;
+
+  &:before {
+    content: "";
+    background: rgba(0, 0, 0, 0.5);
+    position: absolute;
+    width: 100%;
+    height: 100%;
+    top: 0;
+    left: 0;
+    z-index: -1;
+
+    opacity: 0;
+    transition: opacity 0.2s ease;
+  }
+}
+
+#modal.modal--show {
+  transform: translateY(0%);
+  opacity: 1;
+  visibility: visible;
+  transition: all 0s ease;
```

Comment:
You already have `transition` defined on `#modal` selector, is this needed as well? 
Also, does a `0s` transition actually do anything? 🤔